### ISSUE TITLE
consistent-this shouldn't flag up assignment of this to a variable outside current function.

### ISSUE THREAD

**Issue by LukeusMaximus** [eslint/eslint#7153]
**What version are you using?**
v3.4.0
**What did you do?**
```
let someVariable = null

function someFunction() {
    someVariable = this;
    ...
}
```
**What happened?**
ESLint reported that `someVariable` is an unexpected alias of `this` (consistent-this)
**What did you expect to happen?**
ESLint not to flag this up as a contravention of the consistent-this rule because the assignment statement is assigning `this` to a variable outside the current function. Attempts to alias `this` are always an assignment to a variable declared within the current function.

**Comment by eslintbot** [comment]
Hi @LukeusMaximus, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by kaicataldo** [comment]
I know you filled out most of this, but would you mind reporting back with your configuration and the actual output from ESLint? Thanks!

**Comment by LukeusMaximus** [comment]
My .eslintrc is rather large, so here's a [pastebin](http://pastebin.com/JTSwYhGQ) of it.
I have an example file with the following (the eslint-disable ignores rules we don't care about right now):
```
/* eslint-disable no-unused-vars, no-invalid-this */

let holder = null;

function someFunction() {
    holder = this;
}

```
This results in the following output:
```
/home/luke/dev/web/netos-gui/test.js
  6:5  error  Unexpected alias 'holder' for 'this'  consistent-this
```

**Comment by nzakas** [comment]
In your config file, you only have `"consistent-this": "error",`, so the rule is using the default configuration for `this` variables. If you want to use `handler`, then you need to configure the rule as `"consistent-this": ["error", "handler"]`. The rule only knows to check the names that you specify. Please see the documentation here: http://eslint.org/docs/rules/consistent-this

**Comment by LukeusMaximus** [comment]
Apparently you consider assigning to an existing alias declared in the outer scope as valid.
